microsoft/PowerToys
# [Hosts] Fix first entry insert with empty hosts file

## Summary of the Pull Request

Fix dialog not opening when hosts file is empty.
Improved UI when list of entries is empty (with/without filters).

![image](https://github.com/microsoft/PowerToys/assets/25966642/6496dc64-344f-43e8-af47-0409adb7afe9)

![image](https://github.com/microsoft/PowerToys/assets/25966642/a9308f59-8693-45c4-a0a6-76c2da74812a)


## PR Checklist

- [x] **Closes:** #26514


## Detailed Description of the Pull Request / Additional comments


## Validation Steps Performed

_Be sure to make a backup of the hosts file_

- Delete hosts file
- Open Hosts: "No entries in the hosts file" should be displayed
- "New entry" button and "Add an entry" link should both open the new entry dialog
- Add an entry: entry should be added
- Filter entries with a non valid search: "No filter results" should be displayed
- "Clear filters" link should clear the filters
